streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorRecordContext.java

```diff
@@ -73,7 +73,7 @@ public Headers headers() {
         return headers;
     }
 
-    public long sizeBytes() {
+    public long residentMemorySizeEstimate() {
```

Comment:
I should have renamed this method last time I touched this ecosystem. It was confusingly named, since the memory size is closely related to, but different from the serialized format size. I'm just renaming it now, as cleanup.

Reply:
Can you elaborate? Is this not just counting the serialized bytes?

Reply:
no, this method is used for estimating the amount of heap this object occupies. It definitely includes the serialized bytes of the key and value (and headers), but it does not have to include (for example) all the array-size integers we have to add to the serialized format so that we can deserialize them later. These array bounds actually wind up inflating the size of the output quite a bit.